### ISSUE TITLE
Random missing areas get bad effect

### ISSUE THREAD

**Issue by windSong1** [pathak22/context-encoder#22]
hello! Thanks for your brilliant paper and code first.

I use the 'train_random.lua' script to train the net to complete an image with random loss area,I train it in a GPU of GTX 1080, I found the train time is so long if I loop 500 times, almost 150 hours, so I only loop 80 times.the effect in visulization web is good,but when I test it in validation data, the result is so bad.

Followings are some results of my training in random areas,the effet is bad
![screenshot from 2017-11-19 17-28-00](https://user-images.githubusercontent.com/19423217/32989328-075c2c4c-cd4f-11e7-8126-a74f20784973.png)

Followings are resuls from  weight in center missing that you offered for me, the result is good.
![screenshot from 2017-11-19 17-33-34](https://user-images.githubusercontent.com/19423217/32989381-ccfcd97e-cd4f-11e7-8568-e00a6b4ababa.png)

is it the reason that my epoch loops is not enough? any advices will be appreciated.

**Comment by windSong1** [comment]
I got it, I trained it with fineSize of 64x64, but I test it with fineSize of 128x128, but when I use fineSize of 64x64 to test, the result is still not good,I only trained it 80 epoch because of the time.

![screenshot from 2017-11-21 10-59-31](https://user-images.githubusercontent.com/19423217/33052712-18e5c13c-ceab-11e7-80e6-cbd3d0fc6e6d.png)

**Comment by maryam089** [comment]
@linyuxuan you mean results are good 
At training with fine size 64x64 and testing at 128x128 ?

**Comment by windSong1** [comment]
@maryam089 no, I got a mistake at first, as you can see, the result is not good if the train size and test size are not matching. but when I match them all, the resul is still not good, see the last picture. I guess it is the reson that my training epoches are not enough,only 80 epoches.

**Comment by maryam089** [comment]
@windSong1 Well I think so that is the reason but as training on 100K Images from ImageNet takes so long so there is risk that one is wasting time on training for so much days and then end up with same bad results. Anyway could you share your model of 80 epoch with me ? let me check it on some other dataset and see how results changes

**Comment by windSong1** [comment]
@maryam089 I trained it on flying-chair dataset,not ImageNet,do you still want it?

**Comment by micklexqg** [comment]
@windSong1 , I notice that the random position is also fixed from your result image.
if i want to inpaint other areas or anywhere. how should i do ?

**Comment by NerminSalem** [comment]
did u tried to activate adv loss?
